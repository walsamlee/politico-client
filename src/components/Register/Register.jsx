/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';

class Register extends Component {
  constructor(props) {
    super(props);

    this.state = {
      preview: '',
      firstname: '',
      lastname: '',
      othername: '',
      telephone: '',
      email: '',
      password: '',
    };
  }

    registerHandler = (event) => {
      event.preventDefault();

      const userData = {
        passportUrl: this.state.preview,
        firstname: this.state.firstname,
        lastname: this.state.lastname,
        othername: this.state.othername,
        phoneNumber: this.state.telephone,
        email: this.state.email,
        password: this.state.password,
      };

      this.props.registerUser(userData);
    }

    changeHandler = (event) => {
      this.setState({ [event.target.name]: event.target.value });
    }

    imageUploadHandler = (event) => {
      const cloudinaryUrl = 'https://api.cloudinary.com/v1_1/walsam/image/upload';
      const cloudinaryUploadPreset = 'w4rdqtby';

      const file = event.target.files[0];

      const imageFormData = new FormData();

      imageFormData.append('file', file);
      imageFormData.append('upload_preset', cloudinaryUploadPreset);

      fetch(cloudinaryUrl, {
        method: 'POST',
        body: imageFormData,
      })
        .then(response => response.json())
        .then((response) => {
          this.setState({ preview: response.url });
        })
        .catch(error => error);
    }

    render() {
      return (
        <div className="main">
          <div className="dashboard">
            <div className="signup-form">
              <form id="sign-up" name="registration" onSubmit={this.registerHandler}>
                <table className="form-table">
                  <tr>
                    <td className="label" />
                    <td>
                      <input id="passporturl" type="file" accept="image/*" name="image" onChange={this.imageUploadHandler} value={this.state.image} />
                    </td>
                  </tr>
                  <tr>
                    <td className="label" />
                    <td>
                      <input id="firstname" type="text" name="firstname" placeholder="First Name" required onChange={this.changeHandler} value={this.state.firstname} />
                    </td>
                  </tr>
                  <tr>
                    <td className="label" />
                    <td>
                      <input id="lastname" type="text" name="lastname" placeholder="Last Name" required onChange={this.changeHandler} value={this.state.lastname} />
                    </td>
                  </tr>
                  <tr>
                    <td className="label" />
                    <td>
                      <input id="othername" type="text" name="othername" placeholder="Other Name" required onChange={this.changeHandler} value={this.state.othername} />
                    </td>
                  </tr>
                  <tr>
                    <td className="label" />
                    <td>
                      <input id="telephone" type="text" name="telephone" placeholder="Phone Number" required onChange={this.changeHandler} value={this.state.telephone} />
                    </td>
                  </tr>
                  <tr>
                    <td className="label" />
                    <td>
                      <input id="email" type="email" name="email" placeholder="Email Address" required onChange={this.changeHandler} value={this.state.email} />
                    </td>
                  </tr>
                  <tr>
                    <td className="label" />
                    <td>
                      <input id="password" type="password" name="password" placeholder="Password" required onChange={this.changeHandler} value={this.state.password} />
                    </td>
                  </tr>
                  <tr>
                    <td />
                    <td>
                      <input type="submit" className="btn btn-cart" value="Register" />
                      <input type="reset" className="btn btn-cart" value="Reset" />
                    </td>
                  </tr>
                </table>
              </form>
            </div>
          </div>
        </div>
      );
    }
}

export default Register;
