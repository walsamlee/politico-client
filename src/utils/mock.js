const voteData = {
  office: 1,
  candidate: 1,
  voter: 1,
};

const data = {
  candidate: {
    candidates: [
      {
        officeid: 4,
        firstname: 'Admin',
        lastname: 'Admin',
        othername: 'Sam',
        passporturl: 'https://res.cloudinary.com/walsam/image/upload/v1559164438/IMG_5704.jpg',
        userid: 1,
        partyid: 9,
        name: 'Legislative',
        logourl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
      },
      {
        officeid: 6,
        firstname: 'Hassan',
        lastname: 'Hadji',
        othername: 'Chikaso',
        passporturl: '',
        userid: 2,
        partyid: 8,
        name: 'Chairman',
        logourl: 'https://res.cloudinary.com/walsam/image/upload/v1550612284/zausu6uc1iet2lo4fsqr.png',
      },
    ],
  },
  getOffices: {
    offices: [],
  },
  parties: {
    parties: [],
  },
};

const details = {
  office: 1,
  candidate: 2,
};

const token = '';

const login = {
  login: {
    token: '',
    user: [
      {
        user: {
          isAdmin: '',
          id: 1,
          passportUrl: '',
          email: '',
          name: '',
          phoneNumber: 14141414441,
        },
      },
    ],
  },
};

const partiesData = {
  parties: {
    fetching: '',
    fetched: '',
    parties: [
      {
        logoUrl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
        name: 'wrrryery',
        id: 1,
        fetching: true,
      },
      {
        logoUrl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
        name: 'uiyt',
        id: 2,
        fetching: true,
      },
    ],
  },
};

const partyData = {
  id: '',
  image: '',
  partyname: '',
  parthqaddress: '',
};

const profileData = {
  picture: '',
  name: '',
  email: '',
  phone: '',
};

const registerData = {
  preview: '',
  firstname: '',
  lastname: '',
  othername: '',
  telephone: '',
  email: '',
  password: '',
};

const candidate = {
  passporturl: '',
  firstname: '',
  lastname: '',
  name: '',
  logourl: '',
};

const party = {
  id: '',
  name: '',
};

const aspirantData = {
  id: '',
  office: '',
  party: '',
};

const setPartyState = [
  {
    id: 1,
    logoUrl: '',
    name: '',
    hqAddress: '',
  },
];

const mock = {
  voteData,
  data,
  details,
  token,
  login,
  partiesData,
  partyData,
  profileData,
  registerData,
  candidate,
  party,
  aspirantData,
  setPartyState,
};

export default mock;
