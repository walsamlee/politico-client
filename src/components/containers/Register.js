import { connect } from 'react-redux';
import authAction from '../../actions/authAction';
import Register from '../Register/Register';

const mapStateToProps = state => ({
  data: state,
});

const mapDispatchToProps = dispatch => ({
  registerUser: (userData) => {
    dispatch(authAction.registerUser(userData));
  },
});

const AuthRegister = connect(
  mapStateToProps,
  mapDispatchToProps,
)(Register);

export default AuthRegister;
