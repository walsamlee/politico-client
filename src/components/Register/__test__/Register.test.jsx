import React from 'react';
import { shallow } from 'enzyme';
import mock from '../../../utils/mock';
import Register from '../Register';

const { registerData } = mock;

describe('<Register />', () => {
  it('should render Register component', () => {
    const wrapper = shallow(
      <Register userData={registerData} registerUser={jest.fn()} imageFormData={jest.fn()} />,
    );
    const event = {
      preventDefault: jest.fn(),
      target: {
        name: '',
        value: '',
        files: [],
      },
    };
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div'));
    wrapper.instance().registerHandler(event);
    wrapper.instance().changeHandler(event);
  });
});
