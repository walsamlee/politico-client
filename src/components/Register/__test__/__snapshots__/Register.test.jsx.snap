// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> should render Register component 1`] = `ShallowWrapper {}`;
